query.c

```diff
@@ -1626,6 +1675,11 @@ query_add_optional(query_type *q, nsd_type *nsd)
 		if (q->edns.dnssec_ok)	edns->ok[7] = 0x80;
 		else			edns->ok[7] = 0x00;
 		buffer_write(q->packet, edns->ok, OPT_LEN);
+
+		/* Add Extended DNS Error (RFC8914) to verify that we stay in bounds */
+		if (q->edns.ede >= 0)
+			q->edns.opt_reserved_space += 6 + q->edns.ede_text_len - 1;
```

Comment:
Voor als `ede_text_len == 0`, als er dus geen tekst is:
```suggestion
			q->edns.opt_reserved_space += 6 + ( q->edns.ede_text_len
			                                  ? q->edns.ede_text_len - 1 : 0);
```